src/components/SingleEvent/EventFilters.js

```diff
@@ -90,22 +113,34 @@ class EventFilters extends Component {
           wide
         />
         {enableQrCodeScanner ? (
-          <SafeQuery query={QR_SUPPORTED_QUERY}>
-            {({ data = {} }) => {
-              return data.supported ? (
-                <ApolloConsumer>
-                  {client => (
-                    <QRCodeContainer>
-                      <Button onClick={this._scan(client)}>Scan QRCode</Button>
-                      {scanError ? (
-                        <WarningBox>{`${scanError}`}</WarningBox>
-                      ) : null}
-                    </QRCodeContainer>
-                  )}
-                </ApolloConsumer>
-              ) : null
-            }}
-          </SafeQuery>
+          <QRCodeContainer>
+            <Button
+              onClick={() => this.setState({ scannerOn: !scannerOn })}
+              disabled={cameraUnavailable}
+            >
+              {' '}
+              {scannerOn ? 'Stop Scanning' : 'Scan QRCode'}
+            </Button>
+            {scanError ? <WarningBox>{`${scanError}`}</WarningBox> : null}
+            {this.state.scannerOn ? (
+              <QRScannerContainer>
+                <CenteredQrReader
+                  delay={400} // delay = false stops scanning
+                  onError={err => {
+                    console.log(err)
+                    this.setState({
+                      scannerOn: false,
+                      scanError: 'FATAL ERROR'
+                    })
+                    this._disableCamera()
+                    setTimeout(() => this.setState({ scanError: false }), 3000)
```

Comment:
Can we add a comment to explain the logic behind this timeout? Seems like the error just pops up 3 seconds and then disappears? If so we can also add a variable to explain 3000, such as `const TIME_ERROR_APPEARS` or something similar

Reply:
Sure thing

> @TomAFrench How difficult to add a feature to show warning message if certain browsers do not allow camera access?

Not hard at all. I'll add it